ports/rp2/mphalport.h

```diff
@@ -126,4 +126,24 @@ static inline void mp_hal_pin_high(mp_hal_pin_obj_t pin) {
     gpio_set_mask(1 << pin);
 }
 
+enum mp_hal_pin_interrupt_trigger {
+    MP_HAL_PIN_TRIGGER_LOW = GPIO_IRQ_LEVEL_LOW,
+    MP_HAL_PIN_TRIGGER_HIGH = GPIO_IRQ_LEVEL_HIGH,
+    MP_HAL_PIN_TRIGGER_FALL = GPIO_IRQ_EDGE_FALL,
+    MP_HAL_PIN_TRIGGER_RISE = GPIO_IRQ_EDGE_RISE,
+    MP_HAL_PIN_TRIGGER_NONE
+};
+
+extern void machine_pin_irq_helper(mp_hal_pin_obj_t pin, mp_obj_t handler, mp_uint_t trigger, bool hard);
+#define mp_hal_pin_interrupt machine_pin_irq_helper
```

Comment:
I don't think this define is needed.  Also the `extern` is not needed on the declaration.